### PR TITLE
Reporting

### PR DESCRIPTION
**Pull request by AnatoleAM**
Adding queries and mutations for user/emote reporting

### CODE REVIEW

File: src/configure/collections.go [hunk 1]
```diff
@@ -2,19 +2,25 @@ package configure
...
-	CollectionNameBadges            mongo.CollectionName = "badges"
-	CollectionNameRoles             mongo.CollectionName = "roles"
-	CollectionNameAudit             mongo.CollectionName = "audit"
-	CollectionNameEntitlements      mongo.CollectionName = "entitlements"
-	CollectionNameNotifications     mongo.CollectionName = "notifications"
-	CollectionNameNotificationsRead mongo.CollectionName = "notifications_read"
-)
-
-var Indexes = []mongo.IndexRef{}
+var Indexes = []mongo.IndexRef{
```
Comment (TroyDota) [inline src/configure/collections.go:10]: why is this a global variable?

File: src/server/v3/schema/reports.gql [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+enum TargetKind {
+	EMOTE
+	USER
+}
+
+enum ReportStatus {
+	OPEN
+	ASSIGNED
+	CLOSED
+}
```
Comment (TroyDota) [inline src/server/v3/schema/reports.gql:24]: add new line at end of file.

**Review by TroyDota — CHANGES_REQUESTED** [review]
Can you walk me through the design and why you made the choices you did?

**Comment by AnatoleAM** [comment]
@TroyDota, on the design:

Reports are structured with a `target_kind` value denoting the type of object being reported (i.e a user or an emote). The report contains two text fields, a `subject` field which contains the complaint (i.e "My emote was stolen") and `body` which allows adding extra information, such as evidence.

Reports can also have assignees (i.e a moderator) and a priority (for example to give special priority to some users). 
Furthermore this implements a "notes" system, which allows quick communication with the reporter.

File: src/configure/collections.go (hunk 1 above)
Comment (AnatoleAM) [reply src/configure/collections.go:10]: Referenced in `main.go` to set up indexes

**Pull request merged**